src/vs/editor/standalone/browser/standaloneLanguages.ts

```diff
@@ -40,17 +40,17 @@ export function getLanguages(): ILanguageExtensionPoint[] {
 }
 
 export function getEncodedLanguageId(languageId: string): number {
-	let lid = StaticServices.modeService.get().getLanguageIdentifier(languageId);
-	return lid ? lid.id : 0;
+	const modeService = StaticServices.modeService.get();
+	return modeService.languageIdCodec.encodeLanguageId(languageId);
 }
 
 /**
  * An event emitted when a language is first time needed (e.g. a model has it set).
  * @event
  */
 export function onLanguage(languageId: string, callback: () => void): IDisposable {
-	let disposable = StaticServices.modeService.get().onDidEncounterLanguage((languageIdentifier) => {
-		if (languageIdentifier.language === languageId) {
+	let disposable = StaticServices.modeService.get().onDidEncounterLanguage((languageId) => {
+		if (languageId === languageId) {
```

Comment:
This change is incorrect. The condition is now always true. https://github.com/microsoft/monaco-editor/issues/2750

Reply:
Thank you for pointing this out!